src/main/java/cz/cvut/kbss/termit/persistence/dao/skos/SKOSImporter.java

```diff
@@ -100,74 +123,134 @@ private void parseDataFromStreams(String mediaType, InputStream... inputStreams)
+        final Set<Statement> labels = model.filter(Values.iri(glossaryIri), DCTERMS.TITLE, null);
+        labels.stream().filter(s -> {
+            assert s.getObject() instanceof Literal;
+            return Objects.equals(config.get(ConfigParam.LANGUAGE),
+                ((Literal) s.getObject()).getLanguage().orElse(config.get(ConfigParam.LANGUAGE)));
+        }).findAny().ifPresent(s -> vocabulary.setLabel(s.getObject().stringValue()));
+    }
+
+    private Vocabulary createVocabulary(boolean rename) {
+        final String newVocabularyIriBase = resolveVocabularyIriFromImportedData(model);
+        String newVocabularyIri = getFreshVocabularyIri(rename, newVocabularyIriBase);
+        final Vocabulary vocabulary = new Vocabulary();
+        vocabulary.setUri(URI.create(newVocabularyIri));
+
+        String newGlossaryIri = getFreshGlossaryIri(rename, newVocabularyIri);
+        return setupVocabulary(vocabulary, newGlossaryIri);
+    }
+
+    private Vocabulary getVocabulary(boolean rename, final URI vocabularyIri) {
+        final Vocabulary vocabulary = vocabularyDao.find(vocabularyIri).get();
+
+        final String newGlossaryIri = getFreshGlossaryIri(rename, vocabularyIri.toString());
+        return setupVocabulary(vocabulary, newGlossaryIri);
+    }
+
+    private Vocabulary setupVocabulary(final Vocabulary vocabulary, final String glossaryIri) {
+        final Glossary glossary = new Glossary();
+        glossary.setUri(URI.create(glossaryIri));
+        vocabulary.setGlossary(glossary);
+        vocabulary.setModel(new cz.cvut.kbss.termit.model.Model());
+        setVocabularyLabelFromGlossary(vocabulary, glossaryIri);
+        return vocabulary;
+    }
+
+    private String getFreshVocabularyIri(final boolean rename, final String newVocabularyIriBase) {
+        final String newVocabularyIri;
+        if (rename) {
+            newVocabularyIri = getUniqueIriFromBase(newVocabularyIriBase, (r) -> vocabularyDao.find(URI.create(r)));
+            if (!newVocabularyIri.equals(newVocabularyIriBase)) {
+                Utils.changeNamespace(newVocabularyIriBase, newVocabularyIri, model);
+            }
+        } else {
+            newVocabularyIri = newVocabularyIriBase;
+        }
+
+        final Optional<Vocabulary> possiblyVocabulary = vocabularyDao.find(URI.create(newVocabularyIri));
+        if (possiblyVocabulary.isPresent()) {
+            throw new IllegalArgumentException("The same vocabulary already exists elsewhere, set rename=true if you don't mind that the glossary IRI changes.");
+        }
+
+        return newVocabularyIri;
+    }
+
+    private String getFreshGlossaryIri(final boolean rename, final String newVocabularyIri) {
+        final String newGlossaryIri;
+        if (rename) {
+            newGlossaryIri = getUniqueIriFromBase(newVocabularyIri + "/glosář", (r) -> vocabularyDao.findGlossary(URI.create(r)));
```

Comment:
Hardcoded `/glosář` glossary IRI suffix. What about deployments in other languages?